honeybee_vtk/file_writers.py

```diff
@@ -0,0 +1,229 @@
+            file name for the .zip file.
+        target_folder: A text string to a folder to write the output file. The file
+            will be written to the current folder if not provided.
+        include_grids: A boolean. Defaults to True. Grids will not be extracted from
+            HBJSON if set to False.
+        include_vectors: A boolean. Defaults to True. Vector arrows will not be created
+            if set to False.
+        vtk_writer: A VTK writer object.
+        vtk_extension: A text string for the file extension to be used.
+        hbjson: A dictionary.
+
+    Returns:
+        A text string containing the path to the .zip file with VTK/VTP files.
+    """
+    # Get points and face_types from HBJSON
+    points, hb_types = read_hbjson(hbjson)
+
+    # Get points grouped for each Honeybee face_type
+    grouped_points = group_by_face_type(points, hb_types)
+
+    # Write VTK files based on Honeybee face_type and Honeybee object type
+    for hb_type in grouped_points:
+        write_polydata(grouped_points[hb_type], hb_type, vtk_writer, vtk_extension,
+                       target_folder)
+
+    # Names of VTK files to be written
+    file_names = list(grouped_points.keys())
+
+    # write grid points to a vtk file if grids are found in HBJSON
+    grids = check_grid(hbjson)
+
+    # If grids are there in HBJSON and they are requested
+    if grids and include_grids:
+        grid_file_names = _write_grids(grids, vtk_writer, vtk_extension, target_folder)
+        file_names.extend(grid_file_names)
+
+    # Write vectors if they are requested
+    if include_vectors:
+        to_write_vectors = [hb_types, grouped_points, grids, include_grids, vtk_writer,
+                            vtk_extension, target_folder]
+        vector_file_names = _write_vectors(*to_write_vectors)
+        file_names.extend(vector_file_names)
+
+    # Use the name of HBJSON if file name is not provided
+    if not file_name:
+        name = '.'.join(os.path.basename(file_path).split('.')[:-1])
+        file_name = clean_string(name)
+
+    # remove extension if provided by user
+    file_name = file_name if not file_name.lower().endswith('.zip') else file_name[:-4]
+    
+    # Create a .zip file to capture all the generated .vtk files
+    zip_file = os.path.join(target_folder, file_name + '.zip')
+    zipobj = ZipFile(zip_file, 'w')
+
+    # Capture vtk files in a zip file.
+    os.chdir(target_folder)
+    for file_name in file_names:
+        zipobj.write(file_name + vtk_extension)
+    zipobj.close()
+
+    # Delete vtk files if there is permission
+    for file_name in file_names:
+        try:
+            file_name = os.path.join(target_folder, file_name + vtk_extension)
+            os.remove(file_name)
+        except OSError:
+            warnings.warn(
+                f'Honeybee does not have permission to delete {file_names}.'
+                ' You may please delete the .vtk files manually.'
+            )
```

Comment:
Now that we have the basic workflow working I have two comments for improvements. We don't have to address them in this PR but it will be good to address them at some point:

1. Instead of writing the file and deleting them from the target folder the common practice is to use a temp folder. Python has a temfile module that can help you with that. You can create a temp folder and create the files there and then copy what you need to target folder and don't worry about deleting the files yourself or permission issues. Here is a good example: https://stackoverflow.com/a/55104228
2. Write now the zip file includes the full path to the files that it puts in the zip file. It's not desired. What you really want is to have the files in the higher level. Here is an example file that is created by honeybee-vtk. 
[waeanesa_north.zip](https://github.com/ladybug-tools/honeybee-vtk/files/5846135/waeanesa_north.zip)

Reply:
I like this idea of temp folder. I myself did not like creating and deleting files on a folder a user would assign. I will implement this.
This path inside the zip is something that I didn't like as well and that I have already fixed.